### DISCUSSION TITLE
Main frontpage and Tags

### DISCUSSION THREAD

**Discussion by GeneralAccess** [community/community#35612] [New to GitHub] [answered]
This is a newbie question....I think I've understood about branches and tags. I have only one branch - Main - with a few tags for different releases. Why does the first page of my repository always show a really old version, not the most recent tag? That looks out-of-date and I want to link to a file on that page - which will be the wrong version. Is that just the way GitHub works, or is there something I can do to update it?

**Comment by airtower-luna** [comment, marked as answer]
> Why does the first page of my repository always show a really old version, not the most recent tag?

Where exactly does it show that? Do you mean the "releases" section, or something else?

**Reply by GeneralAccess** [reply]
Go to repositories then click on a repository, so you are on the top level page for a particular repository, for example 
https://github.com/GNOME/gedit

**Reply by airtower-luna** [reply]
Yes, but where do you see the old version, and of what? Are you referring to the listed files or something else?

**Reply by GeneralAccess** [reply]
The listed files.

**Reply by airtower-luna** [reply]
Okay, in that case: The repository listing always shows the *default branch* of the repository. If you have only one branch that'll be the default branch, make sure to push any updates there, too, and not only to tags.

**Reply by GeneralAccess** [reply]
That makes sense, I'll give it a go. Thanks!

**Reply by GeneralAccess** [reply]
> ...the default branch...make sure to push any updates there, too, and not only to tags...

I see the point now, I should have been 'push'ing to the default branch, the 'tags' are just snapshots and so are optional. I've done that and it works fine. Thanks for reading my question thoroughly and explaining the answer in simple terms.

**Reply by airtower-luna** [reply]
> I should have been 'push'ing to the default branch, the 'tags' are just snapshots and so are optional.

Yes, the point of branches is to handle ongoing development (like a branch on a tree grows), while tags mark a fixed state of a repository, so they're good to say "this is version X".

Glad I could help! Marking the answer would be appreciated. :slightly_smiling_face:

**Reply by PopoutApps** [reply]
It's often the basic concepts which are the hardest to discover. I hope the thumbs-up and the smiley are what you meant by "marking the answer".

**Reply by airtower-luna** [reply]
Those are nice too, but I meant this:

![grafik](https://user-images.githubusercontent.com/5814919/194942037-0c304805-a33b-4c0e-b072-83c5bcaf15c4.png)

That's only visible to the person who initially created the discussion (and mods apparently,  I've seen answers selected by them). It lets people know the question has been answered, and is nice recognition for the person who did. :slightly_smiling_face:

**Reply by PopoutApps** [reply]
I was looking for that, but it doesn't show.
![1](https://user-images.githubusercontent.com/77306008/195078826-1af7b244-124d-4f04-86db-9b4e7a959ce4.png)
![2](https://user-images.githubusercontent.com/77306008/195078953-fcc39f2f-c206-4e01-a38b-cfd53bd63aee.png)

**Comment by GeneralAccess** [comment]
I made a mistake with accounts, but wasn't able to delete the wrong comments or even report them and ask for them to be deleted.
In addition the format of this forum is horribly confusing. I've had to ask a request for clarification* as the answer, as all the subsequent responses were subsidiary responses or something so couldn't be marked as the answer.
* "Where exactly does it show that? Do you mean the "releases" section, or something else?"

**Reply by airtower-luna** [reply]
Yeah, only being able to mark a top-level response is an issue. It also causes trouble if multiple people contributed to the solution and you want to credit them all.
